rdt/transformers/base.py

```diff
@@ -147,9 +147,9 @@ def _set_columns_data(data, columns_data, columns):
         if isinstance(columns_data, (pd.DataFrame, pd.Series)):
             columns_data.index = data.index
 
-        if len(columns_data.shape) == 1:
+        if columns_data is not None and len(columns_data.shape) == 1:
```

Comment:
Another option is to check if `columns_data` is `None` at the top and just return, so that we don't even bother with the check on line 147